vita3k/renderer/src/gl/sync_state.cpp

```diff
@@ -312,7 +312,51 @@ void sync_texture(GLContext &context, const GxmContextState &state, const MemSta
                 break;
             }
         }
-        renderer::gl::texture::dump(texture, mem, parameter_name, base_path, title_id, program_hash);
+        renderer::gl::texture::dump(texture, mem, parameter_name, base_path, title_id, "fragment", program_hash);
+    }
+
+    glActiveTexture(GL_TEXTURE0);
+}
+
+void sync_vertex_texture(GLContext &context, const GxmContextState &state, const MemState &mem, std::size_t index,
+    const Config &config, const std::string &base_path, const std::string &title_id) {
+    const SceGxmTexture &texture = state.vertex_textures[index];
+
+    if (texture.data_addr == 0) {
+        LOG_WARN("Texture has null data.");
+        return;
+    }
+
+    const SceGxmTextureFormat format = gxm::get_format(&texture);
+    if (gxm::is_paletted_format(format) && texture.palette_addr == 0) {
+        LOG_WARN("Ignoring null palette texture");
+        return;
+    }
+    glActiveTexture(static_cast<GLenum>(static_cast<std::size_t>(GL_TEXTURE0) + index));
```

Comment:
Won't work, you have to rebind your vertex texture indices so that they do not overlap fragment texture indices. e.g vertex texture index 0 cannot have a different texture than fragment texture index 0 in GL because the texture indices are the hw resource being shared. You should remap to something like GL_TEXTURE16 + index (assuming 16 fragment texture slots in vita hw) This is what I meant by extra considerations on shader validation in our previous chat.

Reply:
i see 🤔

Reply:
FYI this also applies to shader bindings, image units cannot overlap or you get invalid shader on compliant drivers. The shader may compile and link but mysteriously fail to render textures on some hw due to TIU aliasing rules. See https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glValidateProgram.xhtml

Reply:
I read that the maximum texture unit 16 in opengl. Is GL_TEXTURE16 + index going to be fine even in binding process?